### DISCUSSION TITLE
Top Sources data delayed?

### DISCUSSION THREAD

**Discussion by bfencken** [plausible/analytics#597] [Self-Hosted Support] [answered]
Turned on a trial for a few of our web properties and despite hitting our site from YouTube, LinkedIn, Twitter and Facebook pages, no referring site sources are showing up despite it being hours since doing so. Is this only refreshed daily?

![image](https://user-images.githubusercontent.com/54144371/104490210-ccd20e00-558d-11eb-9836-c8a8f6ba4ecb.png)

**Comment by metmarkosaric** [comment, marked as answer]
thanks for reaching out @bfencken! this is you trying to visit your own site multiple times per day from different sources? you would only count once per day (the first time) in Plausible as other times you wouldn't be a unique visitor anymore and we only count referral sources that start a new session.

**Comment by bfencken** [comment]
If I use separate browser sessions on Firefox, Chrome and Edge to access the target website from a link posted on another site (YouTube, Twitter, Facebook, Github, etc.) and the separate "visits" count with page views, but I dont see any of the originating source of the traffic.

**Reply by bfencken** [reply]
![image](https://user-images.githubusercontent.com/54144371/104510984-44616680-55a9-11eb-9d67-c45489b77896.png)

**Reply by metmarkosaric** [reply]
I see now. Have you tagged those links with UTM? I ask because "Source" alongside "Medium" and "Campaign" are for UTM tags only. You need to go into "All" in the Top Sources to see all the normal referral sources that are sent via the referral header. Do you see them there?

And you can learn more about how and when to use UTM tags here https://plausible.io/blog/utm-tracking-tags

**Reply by bfencken** [reply]
Make sense. Wasn't aware of the UTM setup. So if I want to post a link to our website on our Twitter profile (or elsewhere) it would be: www.codefactory.software?utm_source=Twitter&utm_medium=Twitter&utm_campaign=Twitter 
(seems clunky) or making a bitly-shortened URL, but those aren't great either. Thanks Marko :-)

**Reply by metmarkosaric** [reply]
You don't need to do it everywhere if you don't want as most sites will send some referral header by default but it certainly helps get more information. For instance, Twitter doesn't tell in the referral header what exact tweet the click came from so by using UTM you could know which clicks came from your Twitter profile link, which came from a specific tweet etc. You can learn even more here: https://plausible.io/docs/manual-link-tagging

**Reply by olevold** [reply]
I've never seen anything in the Top Sources window. I only get this:
![bilde](https://user-images.githubusercontent.com/33154449/104714448-80b9d180-5725-11eb-9bc6-4ee41a011c04.png)

**Reply by metmarkosaric** [reply]
thanks for sharing @olevold! it's really difficult to say much without knowing the URL of your site. is this a self-hosted site or?

**Reply by olevold** [reply]
The site is on AWS (cloudfront) but it's restricted to a specific IP-range for now. Does plausible need access to the pages it tracks?

**Reply by metmarkosaric** [reply]
Plausible script needs to be on each and every page you want to track.

**Reply by olevold** [reply]
I realise that. As you can see, pages are being tracked, but there are no sources. Shouldn't there at least be a something under "Direct / none"? I have also tried query parameter `utm_source`

**Reply by metmarkosaric** [reply]
yes, there normally is a referral source for each visitor you see in the dashboard but something is going wrong in this case. It's really difficult for me to troubleshoot this as it's selfhosted so depends on so many things on your end. let's see if someone else has a better idea

**Reply by olevold** [reply]
I'm not self hosting plausible, if that's what you meant, it's paid for. The site being tracked is _____________

**Reply by metmarkosaric** [reply]
ahh ok you're writing in the self-hosted community support so that's why i understood it as self-hosted. for other support, email is best. i just tried to visit your site and got an error. see:
![Screenshot from 2021-01-15 13-19-22](https://user-images.githubusercontent.com/34340819/104726740-705d2300-5734-11eb-9853-4248c75d7dc1.png)

**Reply by olevold** [reply]
Yes, it's restricted. Sorry, I didn't realise this was the self-hosting forum. I'll send an email

**Reply by metmarkosaric** [reply]
np, thanks!

**Comment by metmarkosaric** [comment]
hey @bfencken! would you be able to mark this thread as "answered" please so it shows up with the response in case others find it in the future? thanks!
